test/controllers/concerns/hal_actions_test.rb

```diff
@@ -1,29 +1,122 @@
 describe HalActions do
-  class HalActionsTestController < ActionController::Base
+
+  class Foo
+    include ActiveModel::Model
+
+    attr_accessor :id, :is_root_resource, :updated_at
+    cattr_accessor :_id
+
+    self._id = 1
+
+    def self.find(*args)
+      Foo.new.tap { |f|
+        f.id = _id
+        self._id += 1
+        f.updated_at = DateTime.parse('1970-01-01 00:01:00')
+      }
+    end
+
+    def self.inject(*args, &block)
+      self._id = 1
+      [find, find].inject(*args, &block)
+    end
+
+    def self.order(*args); self; end
```

Comment:
Unused method argument - `args`. If it's necessary, use `_` or `_args` as an argument name to indicate that it won't be used. You can also write as `order(*)` if you want the method to accept any arguments but don't care about them.